### ISSUE TITLE
<Cluster> | <Service Name> already exits

### ISSUE THREAD

**Issue by singlewind** [awslabs/ecs-refarch-continuous-deployment#11]
In the Deploy stage of my CD pipeline creates a TaskDefinition and Service in the ECS. I expect it recreates a TaskDefinition and update the Service when a new image has been pushed to ECR. However, the TaskDefinition was created but the Service failed to create and update because of the error.

 <Cluster> | <Service Name> already exits

I haven't specify Service Name in the template. Just leave the Stack to figure out. So the Service Name is like <Container Name>-Service-XXXXXXX. Not sure what's wrong with it.

**Comment by jpignata** [comment]
Hmm, I'm unable to reproduce this. The CloudFormation step should create a new task definition and update the service, not attempt to recreate it. Is this the standard reference architecture or did you modify it for your purposes?

**Comment by singlewind** [comment]
@jpignata thank you for reply me. Yes, I steal service template from you. I found another guy have exactly same problem like me. I think it happens when the stack have the same name already exists in the Cloudformation I try to update it via pipeline. And system will create another service first before remove the old one to try minimise the interruption of users. Is there a way we can specify the recreation policy on this, like delete the service first and then create. 

Your template doesn't have any issue I think because the stack provision the cluster as well. In my stack it only provision TaskDefinition and Service.

Anyway, I achieved the same thing by delete stack before create it in the pipeline.
